src/goto-programs/goto_trace.h

```diff
@@ -138,6 +136,22 @@ class goto_trace_stept
     full_lhs_value.make_nil();
     cond_expr.make_nil();
   }
+
+  goto_programt::const_targett get_pc() const { return pc; }
+
+  /// We don't have a way of enforcing this, but \p pc_ must *not* be a
+  /// singular iterator!
+  void set_pc(goto_programt::const_targett pc_)
+  {
+    pc = std::move(pc_);
+    is_pc_set = true;
+  }
+
+  bool get_is_pc_set() const { return is_pc_set; }
```

Comment:
If we set in constructor, we must disable the default constructor. This might be fine, but seems bad.
In general, data members should not be const ([see this article](http://www.drdobbs.com/the-problem-with-const-data-members/184403306)) because this disables copy-assignment. It's better to keep them private and mutable and to maintain constness using public accessor methods.